File: UniShare.API/Models/DTOs/Users/UserSummaryDto.cs
namespace UniShare.API.Models.DTOs.Users;

public class UserSummaryDto
{
    public Guid Id { get; set; }
    public string Email { get; set; } = null!;
    public string FullName { get; set; } = null!;
    public string? AvatarUrl { get; set; }
    public decimal ReputationScore { get; set; }
    public int TotalReviews { get; set; }
    public string? SchoolName { get; set; }
    public string? AreaName { get; set; }
}

File: UniShare.APP/lib/features/users/models/user_summary_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_summary_dto.g.dart';

@JsonSerializable()
class UserSummaryDto {
  final String id;
  final String fullName;
  final String? avatarUrl;
  final String? schoolName;
  final String? areaName;
  final double reputationScore;
  final int totalReviews;

  const UserSummaryDto({
    required this.id,
    required this.fullName,
    this.avatarUrl,
    this.schoolName,
    this.areaName,
    required this.reputationScore,
    required this.totalReviews,
  });

  factory UserSummaryDto.fromJson(Map<String, dynamic> json) =>
      _$UserSummaryDtoFromJson(json);

  Map<String, dynamic> toJson() => _$UserSummaryDtoToJson(this);
}
